bridge/client/app/_components/ktb-stage-overview/ktb-stage-overview.component.ts

```diff
@@ -1,93 +1,47 @@
+import { ActivatedRoute } from '@angular/router';
 
 @Component({
-  selector: 'ktb-stage-overview[project]',
+  selector: 'ktb-stage-overview',
   templateUrl: './ktb-stage-overview.component.html',
   styleUrls: ['./ktb-stage-overview.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 
-export class KtbStageOverviewComponent implements OnInit, OnDestroy {
-  public _project?: Project;
+export class KtbStageOverviewComponent {
+  public project?: Project;
   public selectedStage?: Stage;
   public _dataSource = new DtFilterFieldDefaultDataSource();
   public filter: DtFilterArray[] = [];
   private filteredServices: string[] = [];
-  private globalFilter: {[projectName: string]: {services: string[]}} = {};
-  private unfinishedRoots: Root[] = [];
-  private _rootsTimerInterval = 10;
-  private _rootsTimer: Subscription = Subscription.EMPTY;
-  private readonly unsubscribe$ = new Subject<void>();
+  private globalFilter: { [projectName: string]: { services: string[] } } = {};
 
   @Output() selectedStageChange: EventEmitter<{ stage: Stage, filterType?: string }> = new EventEmitter();
   @Output() filterChange: EventEmitter<string[]> = new EventEmitter<string[]>();
 
-  @Input()
-  get project(): Project | undefined {
-    return this._project;
-  }
-
-  set project(project: Project | undefined) {
-    if (this._project !== project) {
-      this._project = project;
-      this.setFilter();
-    }
-  }
-
-  constructor(private dataService: DataService, private apiService: ApiService, private _changeDetectorRef: ChangeDetectorRef) {
-  }
-
-  ngOnInit(): void {
-    this._rootsTimer = timer(0, this._rootsTimerInterval * 1000)
-      .subscribe(() => {
-        if (this.project) {
-          this.dataService.loadRoots(this.project);
-          if (this.unfinishedRoots) {
-            this.unfinishedRoots.forEach(root => {
-              this.dataService.loadRootTraces(root);
-            });
-          }
-        }
-      });
-
-    this.dataService.roots
+  constructor(private dataService: DataService, private apiService: ApiService, private _changeDetectorRef: ChangeDetectorRef, private route: ActivatedRoute) {
+    const project$ = this.route.params
       .pipe(
-        takeUntil(this.unsubscribe$),
-        filter((roots: Root[] | undefined): roots is Root[] => !!roots),
-        tap(roots => {
-            // Set unfinished roots so that the traces for updates can be loaded
-            // Also ignore currently selected root, as this is getting already polled
-            this.unfinishedRoots = roots.filter(root => root && !root.isFinished());
-          }
-        )).subscribe();
-  }
-
-  ngOnDestroy(): void {
-    this.unsubscribe$.next();
-    this._rootsTimer.unsubscribe();
+        map(params => params.projectName),
+        filter(projectName => !!projectName),
+        switchMap(projectName => this.dataService.getProject(projectName))
+      );
+    project$.subscribe(project => {
```

Comment:
It looks like an unsubscribe is missing here. Either add it to the pipe above or here.

Reply:
done